### PR TITLE
[DOC-9991] v24.1.0 alpha.5 release notes

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-9991] v24.1.0 alpha.5 release notes

Publish 4/1

Preview: [releases/v24.1.md](https://deploy-preview-18433--cockroachdb-docs.netlify.app/docs/releases/v24.1.html)

### CODE REVIEW

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+- [Stored procedure]({% link v24.1/stored-procedures.md %}) now support `OUT` and `INOUT` parameter classes. [#120851][#120851]
+- The [PL/pgSQL]({% link v24.1/plpgsql.md %}) `EXIT` and `CONTINUE` statements can now use labels to specify which loop or block is the target. [#120733][#120733]
+
+<h3 id="v24-1-0-alpha-5-operational-changes">Operational changes</h3>
+
+- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logging.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs#log-buffering-for-network-sinks %) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:37]:
```suggestion
- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs#log-buffering-for-network-sinks %) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+- [Stored procedure]({% link v24.1/stored-procedures.md %}) now support `OUT` and `INOUT` parameter classes. [#120851][#120851]
+- The [PL/pgSQL]({% link v24.1/plpgsql.md %}) `EXIT` and `CONTINUE` statements can now use labels to specify which loop or block is the target. [#120733][#120733]
+
+<h3 id="v24-1-0-alpha-5-operational-changes">Operational changes</h3>
+
+- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs#log-buffering-for-network-sinks %) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:37]:
```suggestion
- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs#log-buffering-for-network-sinks %}) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+- [Stored procedure]({% link v24.1/stored-procedures.md %}) now support `OUT` and `INOUT` parameter classes. [#120851][#120851]
+- The [PL/pgSQL]({% link v24.1/plpgsql.md %}) `EXIT` and `CONTINUE` statements can now use labels to specify which loop or block is the target. [#120733][#120733]
+
+<h3 id="v24-1-0-alpha-5-operational-changes">Operational changes</h3>
+
+- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs#log-buffering-for-network-sinks %}) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:37]:
```suggestion
- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs.md %}#log-buffering-for-network-sinks) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) kv.store.admission.provisioned_bandwidth. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. Replace `{bandwidth-bytes/s}` with the desired bytes per second. [#120895][#120895]
+- Removal of the [cluster setting]({% link v24.1/cluster-settings.md %}) `kv.rangefeed.scheduler.enabled`, which was announced in [v24.1.0-alpha.1`](../v24.1.0-alpha.1.html), has been reverted, and the cluster setting is reinstated. [#121164][#121164]
+
+<h3 id="v24-1-0-alpha-5-db-console-changes">DB Console changes</h3>
+
+- In generated statement fingerprints in the DB Console [**Statements** page]({% link v24.1//ui-statements-page.md %}), lists with only literals or placeholders or similar subexpressions are shortened to their first item followed by "__more__". [#120507][#120507]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:60]:
```suggestion
- In generated statement fingerprints in the DB Console [**Statements** page]({% link v24.1/ui-statements-page.md %}), lists with only literals or placeholders or similar subexpressions are shortened to their first item followed by "__more__". [#120507][#120507]
```

Extra slash breaking the build.

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
+## v24.1.0-alpha.5
+
+Release Date: April 1, 2024
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v24-1-0-alpha-5-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- [Changefeeds]({% link v24.1/change-data-capture-overview.md %}) now default to evenly distributing their work across all replicas, including followers, regardless of leaseholder placement. To disable this behavior, set the [cluster setting]({% link v24.1/cluster-settings.md %}) `changefeed.balanced_distribution.enabled` to `false`. If disabled, now use BulkOracle for planning, which distributes work evenly across all replica in the locality filter, including followers if enabled. This is enabled by default with the cluster setting `changefeed.balanced_distribution.enabled`. If disabled, changefeed planning reverts to its previous behavior for distributing work. [#120077][#120077]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:9]: I don't understand the sentence that starts with "If disabled, now use BulkOracle..." From how it reads, it look like you could delete the following:
```suggestion
- [Changefeeds]({% link v24.1/change-data-capture-overview.md %}) now default to evenly distributing their work across all replicas, including followers, regardless of leaseholder placement. To disable this behavior, set the [cluster setting]({% link v24.1/cluster-settings.md %}) `changefeed.balanced_distribution.enabled` to `false`. If disabled, changefeed planning reverts to its previous behavior for distributing work. [#120077][#120077]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+- When [physical cluster replication]({% link v24.1/physical-cluster-replication-overview.md %}) is enabled, the output of the `SHOW VIRTUAL CLUSTER ... WITH REPLICATION STATUS` command now displays replication lag. [#120782][#120782]
+- When [physical cluster replication]({% link v24.1/physical-cluster-replication-overview.md %}) is enabled, the output of the `SHOW VC WITH REPLICATION STATUS to 1` command has changed:
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:11]:
```suggestion
- When [physical cluster replication]({% link v24.1/physical-cluster-replication-overview.md %}) is enabled, the output of the `SHOW VIRTUAL CLUSTER WITH REPLICATION STATUS to 1` command has changed:
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+- A [stored procedure]({% link v24.1/stored-procedures.md %}) can now invoke another stored procedure using a [`CALL` statement]({% link v24.1/call.md %}). [#120674][#120674]
+- You can now use a [`SET TRANSACTION`]({% link v24.1/set-transaction.md %}) statement within a [PL/pgSQL stored procedure]({% link v24.1/plpgsql.md %}) to configure the transaction isolation level, timestamp, or priority, or to set the transaction to read-only. A `SET TRANSACTION` statement must immediately follow a `COMMIT` or `ROLLBACK`, with no other statements or block boundaries between them. [#120456][#120456]
+- The new [session variable]({% link v24.1/session-variables.md %}) `optimizer_use_virtual_computed_column_stats`, when enabled, configures the [cost-based optimizer]({% link v24.1/cost-based-optimizer.md %}) to use [table statistics]({% link v24.1/show-statistics.md %}) on virtual computed columns. [#120668][#120668]
+- An [identity column]({% link v24.1/create-table.md %}#identity-columns) can now drop the Identity constraint and related sequence using the following SQL statement:
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:22]:
```suggestion
- An [identity column]({% link v24.1/create-table.md %}#identity-columns) can now drop the `IDENTITY` constraint and related sequence using the following SQL statement:
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:25]:
```suggestion
    ~~~ sql
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    ALTER TABLE {table_name} ALTER COLUMN {column_name} DROP IDENTITY [IF EXISTS]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:26]:
```suggestion
    ALTER TABLE {table_name} ALTER COLUMN {column_name} DROP IDENTITY [IF EXISTS];
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    ~~~
+
+    Replace `{table_name}` with the name of the table, and `{column_name}` with the name of the column. `IF EXISTS` is optional, and skips the command if the column is not an identity column.  [#119263][#119263]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:29]:
```suggestion
`IF EXISTS` is optional, and skips the command if the column is not an identity column.  [#119263][#119263]
```
For a release note, I would drop that instruction for replacement.

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+- A shared lock that is acquired explicitly using [`SELECT FOR SHARE`]({% link v24.1/select-for-update.md %}) or implicitly by a [read-committed transaction]({% link v24.1/read-committed.md %}), can now be re-acquired with higher strength by either using a [`SELECT FOR UPDATE`]({% link v24.1/select-for-update.md %}) statement or by writing to the key. [#119671][#119671]
+- [Stored procedure]({% link v24.1/stored-procedures.md %}) now support `OUT` and `INOUT` parameter classes. [#120851][#120851]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:32]:
```suggestion
- [Stored procedures]({% link v24.1/stored-procedures.md %}) now support `OUT` and `INOUT` parameter classes. [#120851][#120851]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+- The [PL/pgSQL]({% link v24.1/plpgsql.md %}) `EXIT` and `CONTINUE` statements can now use labels to specify which loop or block is the target. [#120733][#120733]
+
+<h3 id="v24-1-0-alpha-5-operational-changes">Operational changes</h3>
+
+- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs.md %}#log-buffering-for-network-sinks) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:37]:
```suggestion
- You can now enable asynchronous buffering of `file-group` [log sinks]({% link v24.1/configure-logs.md %}) using the `buffering` [configuration options]({% link v24.1/configure-logs.md %}#log-buffering-for-network-sinks) either by default or to an individual `file-group`. The `buffering` configuration option is incompatible with the `buffered-writes` configuration option. To try the `buffering` option, you must set `buffered-writes: false`. Cockroach Labs recommends setting `max-staleness` to `1s` and `flush-trigger-size` to `256KiB`. [#120428][#120428]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+    - `sql.stats.flush.latency`: The latency of attempted SQL statistics flushes to persistent storage, including both successes and failures. [#120709][#120709]
+- The following new [metrics]({% link v24.1/metrics.md %}) track the number and outcome of proxy requests when `kv.dist_sender.proxy.enabled` is set to `true`:
+    - `distsender.rpc.proxy.sent`
+    - `distsender.rpc.proxy.err`
+    - `distsender.rpc.proxy.forward.sent`
+    - `distsender.rpc.proxy.forward.err`
+
+  Cockroach Labs recommends monitoring and alerting on `distsender.rpc.proxy.sent`, because it indicates a possible network partition. [#120239][#120239]
+- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) kv.store.admission.provisioned_bandwidth. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. Replace `{bandwidth-bytes/s}` with the desired bytes per second. [#120895][#120895]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:55]:
```suggestion
- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) `kv.store.admission.provisioned_bandwidth`. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. Replace `{bandwidth-bytes/s}` with the desired bytes per second. [#120895][#120895]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+<h3 id="v24-1-0-alpha-5-db-console-changes">DB Console changes</h3>
+
+- In generated statement fingerprints in the DB Console [**Statements** page]({% link v24.1//ui-statements-page.md %}), lists with only literals or placeholders or similar subexpressions are shortened to their first item followed by "__more__". [#120507][#120507]
+
+<h3 id="v24-1-0-alpha-5-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug introduced in v23.2 that could cause a [PL/pgSQL]({% link v24.1/plpgsql.md %}) routine to return incorrect results when the routine included:
+    1. At least one parameter.
+    1. An `IF` statement with one leak-proof branch and one branch with side effects. [#120451][#120451]
+- Fixed a rare bug where a `BACKUP` command issued shortly after an `ALTER TABLE {table_name} SET (exclude_data_from_backup = true)` could exclude data from an unrelated table from the backup. [#120188][#120188]
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:67]:
```suggestion
- Fixed a rare bug where a `BACKUP` command issued shortly after an [`ALTER TABLE {table_name} SET (exclude_data_from_backup = true)`]({% link v24.1/take-full-and-incremental-backups.md %}#exclude-a-tables-data-from-backups) could exclude data from an unrelated table from the backup. [#120188][#120188]
```

**Review by kathancox — APPROVED** [review]
Left some comments, I couldn't review from the preview, so you may want to double check the render. Otherwise, LGTM.

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+[#120861]: https://github.com/cockroachdb/cockroach/pull/120861
+[#120895]: https://github.com/cockroachdb/cockroach/pull/120895
+[#121023]: https://github.com/cockroachdb/cockroach/pull/121023
+[#121160]: https://github.com/cockroachdb/cockroach/pull/121160
+[#121164]: https://github.com/cockroachdb/cockroach/pull/121164
+[40f4dc382]: https://github.com/cockroachdb/cockroach/commit/40f4dc382
+[688d6abf4]: https://github.com/cockroachdb/cockroach/commit/688d6abf4
+[8eb7ca994]: https://github.com/cockroachdb/cockroach/commit/8eb7ca994
+[ae8043c90]: https://github.com/cockroachdb/cockroach/commit/ae8043c90
+[d8f53d532]: https://github.com/cockroachdb/cockroach/commit/d8f53d532
```
Comment (kathancox) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:120]: Can these be deleted?

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md (hunk 9 above)
Comment (mdlinville) [reply src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:26]: I agree it looks nicer but it's not strictly required for a one line SQL statement, pretty sure.

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:120]:
```suggestion
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md (hunk 9 above)
Comment (kathancox) [reply src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md]: huh, I think you will see syntax highlighting even on one line.

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 17]
```diff
@@ -19,22 +19,22 @@ Release Date: April 1, 2024
...
 
     {% include_cached copy-clipboard.html %}
-    ~~~ shell
-    ALTER TABLE {table_name} ALTER COLUMN {column_name} DROP IDENTITY [IF EXISTS]
+    ~~~ sql
+    ALTER TABLE {table_name} ALTER COLUMN {column_name} DROP IDENTITY [IF EXISTS];
     ~~~
 
-    Replace `{table_name}` with the name of the table, and `{column_name}` with the name of the column. `IF EXISTS` is optional, and skips the command if the column is not an identity column.  [#119263][#119263]
+`IF EXISTS` is optional, and skips the command if the column is not an identity column.  [#119263][#119263]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:29]:
```suggestion
    `IF EXISTS` is optional, and skips the command if the column is not an identity column.  [#119263][#119263]
```

File: src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md [hunk 18]
```diff
@@ -52,19 +52,19 @@ Release Date: April 1, 2024
...
 
   Cockroach Labs recommends monitoring and alerting on `distsender.rpc.proxy.sent`, because it indicates a possible network partition. [#120239][#120239]
-- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) kv.store.admission.provisioned_bandwidth. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. Replace `{bandwidth-bytes/s}` with the desired bytes per second. [#120895][#120895]
+- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) `kv.store.admission.provisioned_bandwidth`. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. Replace `{bandwidth-bytes/s}` with the desired bytes per second. [#120895][#120895]
```
Comment (mdlinville) [inline src/current/_includes/releases/v24.1/v24.1.0-alpha.5.md:55]:
```suggestion
- The `provisioned-rate` field of a node's store specification can no longer be used to add constraints for the disk name or bandwidth. By default, bandwidth is constrained according to the [cluster setting]({% link v24.1/cluster-settings.md %}) `kv.store.admission.provisioned_bandwidth`. To override this setting for a specific node,  the storage specification must contain `provisioned-rate=bandwidth={bandwidth-bytes/s}`. [#120895][#120895]
```

**Pull request merged**